contractcourt/htlc_timeout_resolver.go

```diff
@@ -48,7 +48,24 @@ type htlcTimeoutResolver struct {
 	// account any fees that may have to be paid if it goes on chain.
 	htlcAmt lnwire.MilliSatoshi
 
-	ResolverKit
+	contractResolverKit
+}
+
+// newTimeoutResolver instantiates a new timeout htlc resolver.
+func newTimeoutResolver(res lnwallet.OutgoingHtlcResolution,
+	broadcastHeight uint32, htlcIndex uint64,
+	htlcAmt lnwire.MilliSatoshi, resKit ResolverKit) *htlcTimeoutResolver {
+
+	return &htlcTimeoutResolver{
+		contractResolverKit: contractResolverKit{
```

Comment:
would it make sense to have a `newContractResolver` func which takes the resolver kit and make the channel there? Nit, but we do have this a repeated a few times.

Reply:
Good suggestion, reworked commits to do this. Also renamed `ResolverKit` to `ResolverConfig` which is really what it is now